addons/phone_validation/models/phone_blacklist.py

```diff
@@ -40,15 +45,19 @@ def create(self, values):
             done.add(sanitized)
             to_create.append(dict(value, number=sanitized))
 
-        """ To avoid crash during import due to unique email, return the existing records if any """
-        sql = '''SELECT number, id FROM phone_blacklist WHERE number = ANY(%s)'''
-        numbers = [v['number'] for v in to_create]
-        self._cr.execute(sql, (numbers,))
-        bl_entries = dict(self._cr.fetchall())
-        to_create = [v for v in to_create if v['number'] not in bl_entries]
+        # Get existing phone blacklist (for requested numbers to create)
+        numbers_requested = [tc['number'] for tc in to_create]
+        pb_existing = self.search([('active', 'in', [True, False]), ('number', 'in', numbers_requested)])
 
-        results = super(PhoneBlackList, self).create(to_create)
-        return self.env['phone.blacklist'].browse(bl_entries.values()) | results
+        # Out of existing pb records, activate non-active
+        pb_existing.filtered(lambda pb: not pb.active).write({'active': True})
+
+        # Create new records, while skipping existing_numbers
+        existing_numbers = pb_existing.mapped('number')
+        to_create_filtered = [d for d in to_create if d['number'] not in existing_numbers]
```

Comment:
Why "d" actually? Keep "v"? Or "number"?
```suggestion
        to_create_filtered = [v for v in to_create if v['number'] not in existing_numbers]
```

Reply:
Going in !
```
    to_create = [values['number'] for values in to_create if values['number'] not in existing_numbers]
```

Reply:
Don't really have a good name for that. Since it's list of dicts I went with 'd', why 'v' :sweat_smile: ? 
Maybe I'd call it `tc` as above, as an element of `to_create`, 

In the same file I used list of numbers, that's why I wanted to avoid calling it anything close to `numbers`, to maximize space between those concepts

Reply:
Because it was used before mostly ^^'

"v" for "values" is pretty common in the code base. "values" could work too. It's good if it's not 100% arbitrary at least.

Reply:
`values` great :+1: